Automattic/Edit-Flow
# Fix/date picker localization

## Description

Fixes #457. Instead of trying to parse the localized date (which fails when using `strtotime` or `DateTime::createFromFormat` for certain cases), we're setting the DateTime numerically so we can parse it.

Also, we're adding a z-index to the DatePicker, it was hidden when using the BlockEditor

## Steps to Test

1. In `Settings`, set language to `Dansk`
2. Select a post, and modify a `date` editorial metadata and select a date in May
3. Save the post, and confirm that the date is saved

Previously, a date in the month of May would not save (because `strotime` and `DateTime:createFromFormat` would fail to parse it) when the language was `Dansk`. This was probably impacting other languages as well
